lib/core/CMakeLists.txt

```diff
@@ -2,6 +2,7 @@ set(
   IRODS_LIB_CORE_HEADERS_PUBLIC
   "${CMAKE_CURRENT_SOURCE_DIR}/include/irods/alignPointer.hpp"
   "${CMAKE_CURRENT_SOURCE_DIR}/include/irods/apiHandler.hpp"
+  "${CMAKE_CURRENT_SOURCE_DIR}/include/irods/authenticate.h"
```

Comment:
This is in lib/core because `rc_authenticate_client` isn't a real API, correct? It wraps a real API call.

Reply:
`rcAuthenticate` / `rsAuthenticate` apparently was an API at some point, but its definition has been lost to time (since before iRODS 3.3). Because we have removed the function declarations and the include path does not change (it's `irods/authenticate.h` either way), I thought it best to move it into `lib/core`.

Reply:
Makes sense.

I'm starting to think the .h/.cpp file should live under lib/api just because it aligns with the idea of an API/network call. It's also where the other client-side APIs live.

Do you feel it would be surprising to a new reader of the code to find this in lib/core instead of lib/api?

Reply:
It would think that would be surprising because many if not most of the headers in `lib/api` declare functions and types which map directly to API calls. `rc_authenticate_client` does not map to an API call, it is more like the interface to the auth plugin framework, not the authentication API (which was built for the authentication framework, which exists in `lib/core`).

However, I see your point that to the average user (or new reader) there is no difference. It's just a library call to them and they probably don't care that the authentication plugin framework is doing most of its work on the client side.

Reply:
I'm not proposing we change where things live, yet. I'm mostly trying to determine whether the location of the files is important given that a consumer of the library won't necessarily care. All of the headers will live in /usr/include/irods. As for a person working on the server code, it is simply a matter of getting used to the codebase.

Reply:
After discussion, we decided to leave this in lib/core. Resolving.